image/api_selector.go

```diff
@@ -230,7 +236,15 @@ func (as *APISelector) buildCandidateTags(params *Params) []*tagSet {
 		sort.Strings(ts.Tags)
 	}
 
-	return result
+	for _, ts := range result {
+		for _, tag := range ts.Tags {
+			if strings.Contains(tag, ",") {
+				return result, workererrors.NewWrappedJobAbortError(errors.Errorf("job was aborted because tag \"%v\" contained \",\", this can happen when .travis.yml has a trailing comma", tag))
```

Comment:
It's intended for travis humans -- though it will potentially show up to end users until we have better handling earlier in the chain.